.github/workflows/publish.yml

```diff
@@ -0,0 +1,43 @@
+name: Publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+
+on:
+  push:
+    tags:
+      - '*'
```

Comment:
Do we ever need to push a tag without releasing?

Do we need to disable this in forks? Can we?

Reply:
> Do we ever need to push a tag without releasing?

I don't understand ?

> Do we need to disable this in forks? Can we?

Forks will not have the secret to upload, and should not push tags on their repo, so  think it's fine.

Reply:
> I don't understand ?

Will it ever be a situation where one would do something like `git tag aas-tutorial` or something crazy, where we need to push a tag but it is not for release? If not, then it's fine.

Reply:
Ah ok, I don't think in a general case. And the build would probably fail, since the package version that would be generated by setuptools_scm is by default not a valid version accepted by PyPI (when it contains the commit hash for dev versions).

Reply:
So to make sure the workflow works, I'm thinking that we could run it on push events (not sure if that include merges on master ?), and just skip the upload part if the push is not a tag with:
```
  if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```
Thoughts?

Reply:
Can we just test it once in this PR before merge by adding push/pull request (not sure which one) but comment out the upload part? Is there a reason why the testing needs to be done for every push to master?

Yes, I think "push" includes merge commits.

Reply:
> Is there a reason why the testing needs to be done for every push to master?

Just to be sure it continues to work in the future, and avoid finding issues when pushing a tag. Running it for pushes/merges on master seems a good solution ? (Or it could also be another build for pull requests)

Reply:
How about "push tag only + scheduled cron" instead of blind "push"? And we skip upload on cron.

Reply:
That could work as well. I would prefer push because it would run just after a PR is merged and also if you push directly a README/Changelog update just before a release.

Reply:
Since you have a strong preference to using `push`, let's try it your way and we can revisit if needed. 😄

Reply:
Ok ;), let's try this.